packages/react-charting/src/components/VerticalBarChart/VerticalBarChartRTL.test.tsx

```diff
@@ -0,0 +1,447 @@
+import { DarkTheme } from '@fluentui/theme-samples';
+import { ThemeProvider } from '@fluentui/react';
+
+const pointsWithLine = [
+  {
+    x: 0,
+    y: 10000,
+    legend: 'Oranges',
+    color: DefaultPalette.accent,
+    xAxisCalloutData: '2020/04/30',
+    yAxisCalloutData: '10%',
+    lineData: {
+      y: 7000,
+      yAxisCalloutData: '34%',
+    },
+  },
+  {
+    x: 10000,
+    y: 50000,
+    legend: 'Dogs',
+    color: DefaultPalette.blueDark,
+    xAxisCalloutData: '2020/04/30',
+    yAxisCalloutData: '20%',
+    lineData: {
+      y: 30000,
+    },
+  },
+  {
+    x: 25000,
+    y: 30000,
+    legend: 'Apples',
+    color: DefaultPalette.blueMid,
+    xAxisCalloutData: '2020/04/30',
+    yAxisCalloutData: '37%',
+    lineData: {
+      y: 3000,
+      yAxisCalloutData: '43%',
+    },
+  },
+  {
+    x: 40000,
+    y: 13000,
+    legend: 'Bananas',
+    color: DefaultPalette.blueLight,
+    xAxisCalloutData: '2020/04/30',
+    yAxisCalloutData: '88%',
+  },
+  {
+    x: 52000,
+    y: 43000,
+    legend: 'Giraffes',
+    color: DefaultPalette.blue,
+    xAxisCalloutData: '2020/04/30',
+    yAxisCalloutData: '71%',
+    lineData: {
+      y: 30000,
+    },
+  },
+  {
+    x: 68000,
+    y: 30000,
+    legend: 'Cats',
+    color: DefaultPalette.blueDark,
+    xAxisCalloutData: '2020/04/30',
+    yAxisCalloutData: '40%',
+    lineData: {
+      y: 5000,
+    },
+  },
+  {
+    x: 80000,
+    y: 20000,
+    legend: 'Elephants',
+    color: DefaultPalette.blue,
+    xAxisCalloutData: '2020/04/30',
+    yAxisCalloutData: '87%',
+    lineData: {
+      y: 16000,
+    },
+  },
+  {
+    x: 92000,
+    y: 45000,
+    legend: 'Monkeys',
+    color: DefaultPalette.blueLight,
+    xAxisCalloutData: '2020/04/30',
+    yAxisCalloutData: '33%',
+    lineData: {
+      y: 40000,
+      yAxisCalloutData: '45%',
+    },
+  },
+];
+
+const simplePoints = [
+  {
+    x: 'This is a medium long label. ',
+    y: 3500,
+    color: '#627CEF',
+  },
+  {
+    x: 'This is a long label This is a long label',
+    y: 2500,
+    color: '#C19C00',
+  },
+  {
+    x: 'This label is as long as the previous one',
+    y: 1900,
+    color: '#E650AF',
+  },
+  {
+    x: 'A short label',
+    y: 2800,
+    color: '#0E7878',
+  },
+];
+
+describe('Vertical bar chart rendering', () => {
+  test('Should render the vertical bar chart with numeric x-axis data', () => {
+    const { container } = render(<VerticalBarChart data={chartPoints} />);
+    expect(container).toMatchSnapshot();
+  });
+
+  test('Should render the vertical bar chart with string x-axis data', () => {
+    const { container } = render(<VerticalBarChart data={simplePoints} />);
+    expect(container).toMatchSnapshot();
+  });
+});
+
+describe('Vertical bar chart - Subcomponent bar', () => {
+  test('Should render the bar with the given width', async () => {
+    // Arrange
+    render(<VerticalBarChart data={chartPoints} barWidth={100} />);
```

Comment:
Is it possible to put the promise in beforeEach and afterEach steps. https://legacy.reactjs.org/docs/testing-recipes.html#setup--teardown

Reply:
This might not be possible as the promise is to be put just after rendering the component to wait for the component to update.